packages/core/src/parsers/linkProcessor.js

```diff
@@ -23,28 +39,57 @@ const defaultTagLinkMap = {
  * @param {string} baseUrl
  */
 function convertRelativeLinks(node, cwf, rootPath, baseUrl) {
-  if (!(node.name in defaultTagLinkMap)) {
-    return;
-  }
-
-  const linkAttribName = defaultTagLinkMap[node.name];
-  const resourcePath = node.attribs && node.attribs[linkAttribName];
+  const resourcePath = getResourcePath(node, cwf, rootPath);
   if (!resourcePath) {
     return;
   }
-
   if (path.isAbsolute(resourcePath) || utils.isUrl(resourcePath) || resourcePath.startsWith('#')) {
     // Do not rewrite.
     return;
   }
 
   const cwd = path.dirname(cwf);
   const fullResourcePath = path.join(cwd, resourcePath);
-  const resourcePathFromRoot = utils.ensurePosix(path.relative(rootPath, fullResourcePath));
+  const resourcePathFromRoot = getResourcePathFromRoot(rootPath, fullResourcePath);
 
+  const linkAttribName = defaultTagLinkMap[node.name];
   node.attribs[linkAttribName] = path.posix.join(baseUrl || '/', resourcePathFromRoot);
 }
 
+function validateIntraLink(node, cwf, rootPath, baseUrl) {
+  // checks if the file indicated in the intra link exists
+  const resourcePath = getResourcePath(node, cwf, rootPath);
+
+  if (!resourcePath || utils.isUrl(resourcePath) || resourcePath.startsWith('#')) {
+    return;
+  }
+
+  // only dealing with absolute paths because relative paths were already converted to absolute
+  let fullResourcePath = path.join(rootPath, resourcePath);
+
+  if (baseUrl) {
+    fullResourcePath = fullResourcePath.replace(baseUrl, '');
+  }
+
+  const link = url.parse(fullResourcePath);
+  if (link.hash) {
+    // remove the hash from fullResourcePath (to properly validate the file existence)
+    fullResourcePath = link.path;
+  }
+
+  // currently users would indicate .html extension, but they should indicate .md
+  // as we will automatically convert the .md extension to .html extension
+  if (fullResourcePath.slice(-5) === '.html') {
+    fullResourcePath = `${fullResourcePath.slice(0, fullResourcePath.length - 5)}.md`;
+  }
```

Comment:
Currently, all our docs have .html extension instead of .md extension. Should I manually convert all of them .md extension and remove this piece of code?

Reply:
Thanks for the initial implementation @wxwxwxwx9 

Two types of files can be linked: pages (`Page` model) and assets (raw unprocessed files)

For assets what you're doing here looks good so far 👍
You'll need to make it respect the [ignore config](https://markbind.org/userGuide/siteJsonFile.html#ignore) though

For pages, because the source file may not end up in `_site` (e.g. `xx.md` -> outputs `xx.html` only), you'll need to first check whether the link is referring to a `Page`, if that fails then you could fallback to the checks here for assets

there's some things in `Site` keeping track of these ^

as an example, the conversion here won't work when you have an actual raw html file (or a html file passed through our processing)

Reply:
Hey Zeyu! I have implemented a draft version based on what you have explained to me. I think the implementation has addressed the following:
- respecting the ignore config for assets
- ensuring the Page checks are not done via FS, but done via memory (passed down as argument)

The implementation is not the most elegant / optimized version. I think it can be improved more. But I just wanted to make sure that this version is working correctly at least! 

I'm not sure if I have captured everything because I think I may have misunderstood the problem or your explanation. But maybe this draft version can help to reveal my misunderstanding.

Reply:
Hey Zeyu! I have moved "validateIntraLink" function to after "convertMdOrMbdExtToHtmlExt" as per your recommendation. I have also made some updates based on that. 

I think everything should be working alright now. The only thing left to do is to see how I can make the code cleaner or more efficient. 

May I know your thoughts? (e.g. anything not working as it should, or ideas to make the code cleaner / optimize it)

Thanks so much for your help so far! :-)